### ISSUE TITLE
conflict when transpiling JSX

### ISSUE THREAD

**Issue by FND** [standard-things/esm#96]
Hopefully nobody will ever need this, but I found myself having to transpile JSX on the fly while also importing ES modules via std/esm - which proved to be trickier than expected due to an apparent conflict with babel-register.

So just in case anyone happens to search for JSX on here, this is what I ended up with:

```javascript
// activate JSX support -- NB: this must be done before @std/esm
require("babel-register")({
    extensions: [".jsx"],
    presets: ["es2015"],
    plugins: [
        ["transform-react-jsx", { pragma: "createElement" }]
    ]
});

// activate support for `import`/`export` (required for non-JSX ES modules)
require = require("@std/esm")(module, { esm: "all", cjs: true });
```

Presumably order matters here because both babel-register and std/esm hook into Node's module extensions.

**Comment by tilgovi** [comment]
I'm not sure in which order these transforms end up applied, but I would think you might want `{modules: false}` on the es2015 preset. You may need `cjs: true` for other reasons, but I'm thinking if you set modules false then your JSX files will be left as esm and maybe you don't need toh CJS interop.

**Comment by jdalton** [comment]
Related to #62.

**Comment by FND** [comment]
@tilgovi: Good idea - if I understand this correctly, this will be possible once babel/babel#6248 has been fixed (though order will still matter as otherwise std/esm inevitably chokes on JSX syntax).

Thanks to both of you!

**Comment by FND** [comment]
I'm afraid v0.11.1 appears to have introduced some sort of incompatibility with babel-register - here's a test case:
https://gist.github.com/FND/e957b9410c20f7e3372058104389de47

That works with v0.11.0, but v0.11.1 appears to choke on JSX:
```
$ node index.js
…/code.jsx:5
    <div class="wrapper">
 ^
SyntaxError: Unexpected token
```

I'm afraid that test case is not quite minimal because there are a few things that need explaining:

* for simplicity, the `createElement` function (and thus the console output) does not preserve element hierarchy
* as described above, we first load babel-register for JSX and then @std/esm for ESM syntax
* `import`ing `*.jsx` modules which include ESM syntax works as expected because Babel transpiles these (`component.jsx` exists only to prove that this also works transitively)
* `import`ing non-`*.jsx` modules which include ESM syntax (i.e. `util.js`) fails because @std/esm's `require` is not global (thus the hack in `index.js`)

(I opted not to create a new issue because this seemed closely related to the previous discussion.)

**Comment by jdalton** [comment]
Hi @FND,

This issue was not a bug report. Babel register is still not patched so there's nothing for us to do on our side.
